crates/core/tedge_mapper/src/sm_c8y_mapper/mapper.rs

```diff
@@ -56,6 +56,27 @@ impl CumulocitySoftwareManagementMapper {
 
         Ok(topic_filter)
     }
+
+    pub async fn init_or_drop(&mut self, sub_or_unsub: bool) -> Result<(), anyhow::Error> {
+        let operations = Operations::try_new("/etc/tedge/operations", "c8y")?;
+        let mqtt_topic = CumulocitySoftwareManagementMapper::subscriptions(&operations)?;
+        let config: Config;
+        if sub_or_unsub {
+            info!("De-Initializing tedge sm mapper");
+            config = Config::default()
+                .with_session_name("SM-C8Y-Mapper")
+                .with_clean_session(true)
+                .with_subscriptions(mqtt_topic);
+        } else {
+            info!("Initializing tedge sm mapper");
+            config = Config::default()
+                .with_session_name("SM-C8Y-Mapper")
+                .with_clean_session(false)
+                .with_subscriptions(mqtt_topic);
+        }
+        let _client = Connection::new(&config).await?;
```

Comment:
We could lose messages here if `--init` is called twice.

* Indeed, the messages start to be consumed on connect even if the client doesn't process them. Behind the scene `rumqttc` is acknowledging the messages and they will not be delivered again.
* I think that one needs to implement the `init_session` and `clear_session` inside the `mqtt_channel` to avoid that by disconnecting as soon as the subscriptions are acknowledged.